backend-rust/src/indexer.rs

```diff
@@ -617,8 +620,9 @@ impl ProcessEvent for BlockProcessor {
         let mut new_context = self.current_context.clone();
         PreparedBlock::batch_save(batch, &mut new_context, &mut tx).await?;
         for block in batch {
+            let mut statistics = Statistics::new(block.height);
```

Comment:
We don't rely on anything that is not already available in the preprocessing step, so it would ideally be computed during this step instead of in the processing step.

Reply:
Well we cant. I need access to the transaction

Reply:
The way for me to achieve this would be to alter PreparedBlock to include Statistics.

Do you find that being more clean?